### PR TITLE
Add command set-bubble-transparent

### PR DESCRIPTION
**Pull request by josa42**
I add a command to set the bubble transparent. This can be be mapped to a key. On the next keyup event the bubble is set opaque again.

```
'atom-text-editor:not(.mini)':
  'ctrl': 'linter:set-bubble-transparent'
```

This was discussed here: #608

### CODE REVIEW

File: lib/commands.coffee [hunk 1]
```diff
@@ -15,6 +16,9 @@ class Commands
     return unless activeEditorLinter
     activeEditorLinter.toggleStatus()
 
+  setBubleTransparent: ->
+    @linter.getActiveEditorLinter()?.setBubbleTransparent()
```
Comment (steelbrain) [inline lib/commands.coffee]: how about calling `@linter.views.setBubbleTransparent` directly

File: lib/editor-linter.coffee [hunk 2]
```diff
@@ -25,6 +25,9 @@ class EditorLinter
   toggleStatus: ->
     @setStatus !@_status
 
+  setBubbleTransparent: ->
```
Comment (steelbrain) [inline lib/editor-linter.coffee]: You can just remove this and access views directly

**Comment by steelbrain** [comment]
Would you mind joining us in slack? 
http://atom-slack.herokuapp.com

**Comment by k2b6s9j** [comment]
:+1: We should merge after the code is cleaned up.

**Comment by steelbrain** [comment]
LGTM after the changes.

File: lib/commands.coffee (hunk 1 above)
Comment (josa42) [inline lib/commands.coffee]: Your right. I'll do that

**Comment by josa42** [comment]
I just noticed, when the editor looses focus the bubble stays transparent. So I need a way to handle this.

**Comment by steelbrain** [comment]
do this in the file require area
```coffee
currentWindow = require('remote').getCurrentWindow()
```
and this in `setTransparent`
```js
currentWindow.on 'blur', @setBubbleOpaque
```
and in `setBubbleOpaque`
```js
currentWindow.off 'blue', @setBubbleOpaque
```

More info [here][1] and [here][2].

[1]:https://github.com/atom/electron/blob/master/docs/api/browser-window.md#event-blur
[2]:https://github.com/atom/electron/blob/master/docs/api/remote.md#remotegetcurrentwindow

**Comment by josa42** [comment]
@steelbrain thx

**Comment by josa42** [comment]
@steelbrain Actually in this case, since it is in a view you access the window directly and use `window.addEventListener` and `window.removeEventListener`.

**Comment by josa42** [comment]
I think it should be working as expected now.

**Comment by steelbrain** [comment]
Confirmed

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 * Fix message render for non TextEditor panes (Fixes #610)
 * Make the bubble follow cursor
 * Show a nicer error if linter binary doesn't exist (Fixes #612)
+* Add set-bubble-transparent to set the bubble transparent until the key is released (Fixes #608)
```
Comment (steelbrain) [inline CHANGELOG.md]: How about adding `command` after that command name?

File: lib/commands.coffee [hunk 4]
```diff
@@ -6,6 +6,7 @@ class Commands
     @_subscriptions.add atom.commands.add 'atom-workspace',
       'linter:next-error': @nextError.bind(@)
       'linter:toggle': @toggleLinter.bind(@)
+      'linter:set-bubble-transparent': @setBubleTransparent.bind(@)
```
Comment (AsaAyers) [inline lib/commands.coffee]: Stand alone `@` are discouraged in CoffeeScript. Instead, you can add a fat arrow to `setBubleTransparent` below and you don't have to bind it here.

Comment (AsaAyers) [inline lib/commands.coffee]: For more info see https://github.com/jashkenas/coffeescript/issues/1601

Comment (steelbrain) [inline lib/commands.coffee]: That one was actually my mistake (see the lines above). Noted thanks.

Comment (josa42) [inline lib/commands.coffee]: Oh I didn't know that. Nice!

**Comment by josa42** [comment]
Do you want me to squash the commites?

File: lib/commands.coffee [hunk 5]
```diff
@@ -6,6 +6,7 @@ class Commands
     @_subscriptions.add atom.commands.add 'atom-workspace',
       'linter:next-error': @nextError.bind(@)
       'linter:toggle': @toggleLinter.bind(@)
+      'linter:set-bubble-transparent': @setBubleTransparent
```
Comment (steelbrain) [inline lib/commands.coffee]: Well, he didn't mean that.
he meant
```coffee
'linter:set-bubble-transparent': => @setBubleTransparent
```
and can you also revert the change to fat arrow below

**Comment by steelbrain** [comment]
Squashing commits is not really necessary :)

File: lib/commands.coffee (hunk 5 above)
Comment (AsaAyers) [inline lib/commands.coffee]: @steelbrain is wrong. you did exactly what I meant.

File: lib/commands.coffee [hunk 6]
```diff
@@ -6,6 +6,7 @@ class Commands
     @_subscriptions.add atom.commands.add 'atom-workspace',
       'linter:next-error': @nextError.bind(@)
       'linter:toggle': @toggleLinter.bind(@)
+      'linter:set-bubble-transparent': => @setBubleTransparent()
```
Comment (AsaAyers) [inline lib/commands.coffee]: That's fine. Both are valid options but this become less clean if parameters were involved.

```CoffeeScript
    'linter:do-something-with-parameters': (param1, param2, param3) => @something(param1, param2, param3)
```

I favor the fat arrow on the target because it's always clean.

Comment (steelbrain) [inline lib/commands.coffee]: @AsaAyers we will not be passing arguments so this looks good to me

Comment (AsaAyers) [inline lib/commands.coffee]: You're correct while still missing the point. This solution is valid here, where the other solution is valid always.

"always" being, as long as you're using CoffeeScript classes, and subclasses are created through extension and not some bizarre property copying. I could make up a stupid way to break this, but I've never seen real code that would.

Comment (steelbrain) [inline lib/commands.coffee]: Can we discuss this in slack please?

**Comment by steelbrain** [comment]
LGTM

**Comment by steelbrain** [comment]
@josa42 Thanks again. :clap:

**Comment by josa42** [comment]
@steelbrain My pleasure!

**Comment by steelbrain** [comment]
People, I feel like we should add a default keybinding for this. 'cause not all of users are gonna want to modify the keymap.

**Comment by iam4x** [comment]
@steelbrain I'm with you!
